### PR TITLE
Refactor xcconfig structure

### LINKED ISSUE #2326: Configure Apple project for Developer signing
Come with a solution to prevent the developer requirement of having private signing certs installed to build and test the Apple app in Debug.

See this commit: https://github.com/firezone/firezone/pull/2313/commits/c88117814b918a531f5510bea71b7f2bda8cafbd

### LINKED ISSUE #2334: iOS pulls wrong xcconfig in CI
Possibly a cache bug as it nothing seems out of the ordinary.

#2333 should make this easier to debug if not solve it entirely.

IMG_0054

### PR DESCRIPTION
**Pull request by jamilbk**
- Conditionally set Automatic or Manual provisioning from env (Fixes #2326)
- Refactor xcconfig structure to make configuring the project simpler across different envs
- Use consistent parentheses in Makefiles and xcconfigs (Refs #2324) (Fixes #2334)

### CODE REVIEW

**Comment by jamilbk** [comment]
@roop @conectado With this PR we should be back to Automatic code signing.

File: swift/apple/Firezone/xcconfig/prod.xcconfig
Comment (jamilbk) [inline swift/apple/Firezone/xcconfig/prod.xcconfig:1]: @roop I thought this structure might be easier to maintain going forward. There is a tiny bit more duplication, but it's very clear what each variable is in each file this way, and it makes setting the right one per environment easier.

File: swift/apple/Firezone.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -545,9 +545,10 @@
 			buildSettings = {
 				CLANG_ENABLE_MODULES = YES;
 				CODE_SIGN_ENTITLEMENTS = FirezoneNetworkExtension/FirezoneNetworkExtension_iOS.entitlements;
-				CODE_SIGN_IDENTITY = "Apple Distribution: Firezone, Inc. (47R2M6779T)";
-				CODE_SIGN_STYLE = Manual;
+				CODE_SIGN_IDENTITY = "$(CODE_SIGN_IDENTITY)";
+				CODE_SIGN_STYLE = "$(CODE_SIGN_STYLE)";
 				CURRENT_PROJECT_VERSION = 0;
```
Comment (roop) [inline swift/apple/Firezone.xcodeproj/project.pbxproj:550]: The generally accepted way to do this is: 

```
CODE_SIGN_STYLE = "$(inherited)";
CODE_SIGN_IDENTITY = "$(inherited)";
DEVELOPMENT_TEAM = "$(inherited)";
```

but I don't know of any issues with how you're doing it.

File: swift/apple/Firezone/xcconfig/prod.xcconfig
Comment (roop) [reply swift/apple/Firezone/xcconfig/prod.xcconfig:1]: This is much cleaner. Well done. 👍🏻

File: swift/apple/README.md [hunk 2]
```diff
@@ -27,11 +27,11 @@ CI/CD pipeline.
...
+1. Copy an appropriate xcconfig and edit as necessary:
 
    ```bash
-   cp Firezone/xcconfig/Developer.xcconfig.firezone Firezone/xcconfig/Developer.xcconfig
-   vim Firezone/xcconfig/Developer.xcconfig
+   cp Firezone/xcconfig/dev.xcconfig Firezone/xcconfig/config.xcconfig
+   vim Firezone/xcconfig/config.xcconfig
    ```
 
    ```bash
```
Comment (roop) [inline swift/apple/README.md]: I think we should remove the lines about Server.xcconfig from the README.

**Review by roop — APPROVED** [review]
Very nicely done. Thank you.

File: swift/apple/Firezone.xcodeproj/project.pbxproj (hunk 1 above)
Comment (jamilbk) [reply swift/apple/Firezone.xcodeproj/project.pbxproj:550]: Oh, that's what I was looking for, thanks!

File: swift/apple/README.md (hunk 2 above)
Comment (jamilbk) [reply swift/apple/README.md]: Good catch, also updated our constants error message text to mention the new filename.

**Pull request merged**